src/clj/harja/palvelin/integraatiot/api/urakat.clj

```diff
@@ -150,22 +150,54 @@
-                                            (catch Object _
-                                              false))))
-                                urakat)]
+      (let [fn-suodata-urakat-oikeuksilla (fn [urakat]
+                                            (let
+                                              [urakat (konv/vector-mappien-alaviiva->rakenne urakat)
+                                               urakat-suodatettu (into []
+                                                                   (filter (fn [urakka]
+                                                                             ;; Ota tuloksiin mukaan vain urakat, joihin käyttäjällä on oikeus
+                                                                             ;; Validointi heittää slingshot-virheen, jos käyttäjällä ei ole oikeuksia.
+                                                                             (try+
+                                                                               (validointi/tarkista-kayttajan-oikeudet-urakkaan
+                                                                                 db (:id urakka) kayttaja)
+                                                                               true
+                                                                               (catch Object _
+                                                                                 false))))
+                                                                   urakat)]
+                                              urakat-suodatettu))
+            kilometrit 100
+            urakat-sijainilla (hae-urakka-sijainnilla* db {:x x-easting :y y-northing
+                                                           :aloitustoleranssi aloitustoleranssi
+                                                           :maksimitoleranssi maksimitolenranssi
+                                                           :urakkatyyppi urakkatyyppi})
+            ;; Suodata kaikki urakat johon ei ole oikeuksia 
+            urakat-sijainilla (fn-suodata-urakat-oikeuksilla urakat-sijainilla)
+            ;; Jos sijainnilla ei löydy tuloksia ja palauta-lahin-hoitourakka on true, haetaan lähin hoitourakka
+            hae-lahin-hoitourakka? (and
+                                     ;; Kun parametri on true, palautetaan lähin hoitourakka jos sijainnilla ei löydy urakoita
+                                     (= palauta-lahin-hoitourakka "true")
+                                     (empty? urakat-sijainilla))
+            ;; Jos halutaan etsiä lähin hoitourakka
+            urakat (if hae-lahin-hoitourakka?
+                     ;; Palauttaa 50 urakkaa kilometrisäteellä
+                     (q-urakat/hae-lahin-hoidon-alueurakka db {:x x-easting :y y-northing :maksimietaisyys (* kilometrit 1000)})
+                     urakat-sijainilla)
+            ;; Suodata kaikki urakat johon ei ole oikeuksia 
+            urakat-suodatettu (fn-suodata-urakat-oikeuksilla urakat)
```

Comment:
(fn-suodata-urakat-oikeuksilla) kutsutaan nyt joissakin tilanteissa kahteen kertaan. Voisi katsoa, jos rakenteen saisi sellaiseksi, että riittäisi kertakutsuminen.

Reply:
Näin on, korjattu.

https://github.com/finnishtransportagency/harja/pull/3107/commits/4566dc1a3757ea397b9cfbf88c7a30e6e2f9d75f